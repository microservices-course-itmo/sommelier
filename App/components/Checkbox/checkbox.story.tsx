import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { boolean } from '@storybook/addon-knobs'

import Checkbox from './checkbox'
import { BufferView, ThemeDecorator } from '../../config/decorators'

const defaultCheckbox = () => (
  <Checkbox checked={boolean('checked', true)}>White wine</Checkbox>
)

storiesOf('Checkbox', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Default checkbox', defaultCheckbox)
