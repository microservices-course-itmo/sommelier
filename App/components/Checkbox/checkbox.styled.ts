import styled from 'styled-components/native'

export const CheckboxContainer = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
`

export const CheckboxText = styled.Text`
  color: ${({ theme }) => theme.colors.black};
  align-self: auto;
  font-size: 20px;
  font-weight: 500;
  margin-left: 40px;
`

export const UncheckedCB = styled.TouchableOpacity`
  height: 18
  width: 18
  borderRadius: 2
  borderWidth: 1
  borderColor: #757575
  alignItems: center
  justifyContent: center
`

export const CheckedCB = styled.TouchableOpacity`
  height: 18
  width: 18
  alignItems: center
  justifyContent: center
`
