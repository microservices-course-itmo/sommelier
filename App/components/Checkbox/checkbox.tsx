import React, { ReactNode } from 'react'
import { StyleSheet } from 'react-native'
import NativeCheckbox from '@react-native-community/checkbox'
import { CheckboxContainer, CheckboxText } from './checkbox.styled'

type Props = {
  checked: boolean
  children: ReactNode
}

const Checkbox = ({ checked, children }: Props) => (
  <CheckboxContainer>
    <NativeCheckbox
      value={checked}
      tintColors={{ true: '#A22841', false: '#757575' }}
    />
    <CheckboxText>{children}</CheckboxText>
  </CheckboxContainer>
)

export default Checkbox
