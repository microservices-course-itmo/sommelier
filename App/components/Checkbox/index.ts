export { default as Checkbox, CheckBoxProps } from './checkbox'
