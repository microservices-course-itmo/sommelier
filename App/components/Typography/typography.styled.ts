import styled from 'styled-components/native'
import {
  TypographyProps,
  StyledHeaderProps,
  StyledTextProps,
} from './typography.types'

const StyledTypography = styled.Text<TypographyProps>`
  color: ${({ color, theme }) =>
    color ? theme.colors[color] : theme.colors.black};
  font-weight: ${({ bold }) => (bold ? 'bold' : 'normal')};
  font-style: ${({ italic }) => (italic ? 'italic' : 'normal')};
  text-decoration: ${({ crossedOut }) =>
    crossedOut ? 'line-through' : 'none'};
`

export const StyledHeader = styled(StyledTypography)<StyledHeaderProps>`
  font-size: ${({ size, theme }) => theme.headerSizes[size]};
`

export const StyledText = styled(StyledTypography)<StyledTextProps>`
  font-size: ${({ size, theme }) => theme.fontSizes[size]};
`
