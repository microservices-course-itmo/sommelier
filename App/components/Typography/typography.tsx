import React, { ReactNode } from 'react'
import * as Styled from './typography.styled'
import theme from '../../theme/index'
import { ReactElement } from 'react'

type Props = {
  children: ReactNode
  color: string
  bold?: boolean
  italic: boolean
  size?: keyof typeof theme.fontSizes
  crossedOut?: boolean
  fontFamily: string
}

export const H1 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH1
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH1>
)

export const H2 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH2
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH2>
)

export const H3 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH3
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH3>
)

export const H4 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH4
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH4>
)

export const H5 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH5
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH5>
)

export const H6 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH6
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH6>
)

export const Text = ({
  children,
  color,
  bold,
  italic,
  size,
  crossedOut,
  fontFamily,
}: Props): ReactElement => (
  <Styled.StyledText
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
    size={size}
    crossedOut={crossedOut}
  >
    {children}
  </Styled.StyledText>
)
