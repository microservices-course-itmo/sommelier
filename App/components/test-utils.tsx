import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
