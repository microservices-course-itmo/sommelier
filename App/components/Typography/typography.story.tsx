import React from 'react'
import styled from 'styled-components/native'
import { storiesOf } from '@storybook/react-native'
import { boolean, select } from '@storybook/addon-knobs'
import { H1, H2, H3, H4, H5, H6, Text } from './typography'
import { BufferView, ThemeDecorator } from '../../config/decorators'
import theme from '../../theme/index'

const StyledWrapper = styled.View`
  display: flex;
  flex-direction: column;
`

const FONT_SIZES = Object.keys(theme.fontSizes) as Array<
  keyof typeof theme.fontSizes
>
const COLORS = Object.keys(theme.colors) as Array<keyof typeof theme.colors>
const FONTS = Object.keys(theme.fonts) as Array<keyof typeof theme.fonts>

export const list = () => (
  <StyledWrapper>
    <H1 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H1 value
    </H1>
    <H2 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H2 value
    </H2>
    <H3 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H3 value
    </H3>
    <H4 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H4 value
    </H4>
    <H5 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H5 value
    </H5>
    <H6 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H6 value
    </H6>
    <Text
      color='black'
      bold={false}
      italic={false}
      fontFamily='PtSans'
      size='md'
    >
      Text value
    </Text>
  </StyledWrapper>
)

export const knobs = () => (
  <StyledWrapper>
    <H1
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H1 value
    </H1>
    <H2
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H2 value
    </H2>
    <H3
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H3 value
    </H3>
    <H4
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H4 value
    </H4>
    <H5
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H5 value
    </H5>
    <H6
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H6 value
    </H6>
    <Text
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
      size={select('text font-size', FONT_SIZES, FONT_SIZES[0])}
      crossedOut={boolean('cross-out', false)}
    >
      Text value
    </Text>
  </StyledWrapper>
)

storiesOf('Typography', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Typography list', list)

storiesOf('Typography', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Typography knobs', knobs)
