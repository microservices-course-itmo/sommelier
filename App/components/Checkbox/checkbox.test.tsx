import React from 'react'
import { themeRender } from '../test-utils'
import Checkbox from './checkbox'
import { Text } from '../Typography'

describe('Checkbox component', () => {
  it('Component with only necessary props matches snapshot', () => {
    const tree = themeRender(
      <Checkbox checked>
        <Text>abc</Text>
      </Checkbox>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
