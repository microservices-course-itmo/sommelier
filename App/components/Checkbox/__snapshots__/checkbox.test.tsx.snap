// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Component with only necessary props matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      },
    ]
  }
>
  <View
    pointerEvents="auto"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      tintColors={
        Object {
          "false": "#757575",
          "true": "#A22841",
        }
      }
      value={true}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "auto",
          "color": "#000",
          "fontSize": 20,
          "fontWeight": "500",
          "marginLeft": 40,
        },
      ]
    }
  >
    <Text
      color="black"
      size="md"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          },
        ]
      }
    >
      abc
    </Text>
  </Text>
</View>
`;
